examples/alicebob-grant-and-line-by-line-PRE.py

```diff
@@ -1,70 +1,77 @@
 # This is an example of Alice setting a Policy on the NuCypher network.
 # In this example, Alice uses n=1, which is almost always a bad idea.  Don't do it.
 
-# WIP w/ hendrix@8227c4abcb37ee6d27528a13ec22d55ee106107f
+# WIP w/ hendrix@3.0.0
 
 import datetime
 import sys
 
-import requests
-
+from examples.sandbox_resources import SandboxNetworkyStuff
 from nkms.characters import Alice, Bob, Ursula
-from nkms.crypto.kits import MessageKit
-from nkms.crypto.powers import SigningPower, EncryptingPower
+from nkms.data_sources import DataSource
 from nkms.network.node import NetworkyStuff
-from umbral import pre
-
-ALICE = Alice()
-BOB = Bob()
-URSULA = Ursula.from_rest_url(address="https://localhost", port="3550")
-
+import maya
 
-class SandboxNetworkyStuff(NetworkyStuff):
-    def find_ursula(self, contract=None):
-        ursula = Ursula.as_discovered_on_network(dht_port=None, dht_interface=None,
-                                                 rest_address="https://localhost", rest_port=3550,
-                                                 powers_and_keys={
-                                                    SigningPower: URSULA.stamp.as_umbral_pubkey(),
-                                                    EncryptingPower: URSULA.public_key(EncryptingPower)
-                                                 }
-                                                 )
-        response = requests.post("https://localhost:3550/consider_contract", bytes(contract), verify=False)
-        response.was_accepted = True
-        return ursula, response
+# This is already running in another process.
+URSULA = Ursula.from_rest_url(NetworkyStuff(), address="localhost", port=3601)
 
-    def enact_policy(self, ursula, hrac, payload):
-        response = requests.post('{}:{}/kFrag/{}'.format(ursula.rest_address, ursula.rest_port, hrac.hex()),
-                                 payload, verify=False)
-        # TODO: Something useful here and it's probably ready to go down into NetworkyStuff.
-        return response.status_code == 200
+#########
+# Alice #
+#########
 
+network_middleware = SandboxNetworkyStuff([URSULA])
+ALICE = Alice(network_middleware=network_middleware)
 
-networky_stuff = SandboxNetworkyStuff()
-
-policy_end_datetime = datetime.datetime.now() + datetime.timedelta(days=5)
+# Here are our Policy details.
+policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+m = 1
 n = 1
-uri = b"secret/files/and/stuff"
+label = b"secret/files/and/stuff"
 
-# Alice gets on the network and discovers Ursula, presumably from the blockchain.
-ALICE.learn_about_nodes(address="https://localhost", port="3550")
 
-# Alice grants to Bob.
+# Alice gets on the network and, knowing about at least one Ursula,
+# Is able to discover all Ursulas.
+ALICE.network_bootstrap([("localhost", 3601)])
 
-policy = ALICE.grant(BOB, uri, networky_stuff, m=1, n=n,
+# Alice grants to Bob.
+BOB = Bob()
+policy = ALICE.grant(BOB, label, m=m, n=n,
```

Comment:
I think, we still should allow label=None. Maybe in this implementation - still with a _chimney_

Reply:
(it would be `label=''` in such case)